src/frontends/common/include/common/input_model.hpp

```diff
@@ -39,7 +39,7 @@ class FRONTEND_API InputModel {
 public:
     typedef std::shared_ptr<InputModel> Ptr;
 
-    virtual ~InputModel() = default;
+    virtual ~InputModel() = 0;
```

Comment:
maybe just `virtual ~InputModel();` ?

Reply:
@ilya-lavrenov No, I want to make these classes abstract. There is no reason to allow instantiation of FrontEnd, InputModel, or Place classes.
`virtual ~InputModel() = 0;` is way to do classes abstract without making any another method pure virtual and without any changes in derived classes. In input_model.cpp, the destructor of base class is defined as default.